syntax = "proto3";

import "google/protobuf/empty.proto";

import "common.proto";

package services;

message WorkerStatus {
    uint32 id = 1;
    optional string message = 2;
}

service Worker {
    // DISPATCHER -> WORKER: Queue a new task for processing on the worker
    rpc receive_task(services.Task) returns (google.protobuf.Empty) {}
    // Any -> WORKER: Query the status of the worker
    rpc get_status(google.protobuf.Empty) returns (WorkerStatus) {}
}